### ISSUE TITLE
Don't throw general exception of the placeOrder

### ISSUE THREAD

**Issue by rikterbeek** [magento/payment-improvements#21]
PaymentProviders can have specific reasons why a placeOrder can fail. We provide this error back as an exception but that message can never be shown to the front-end because the core catched the exception and changed it to a generic CouldNotSaveException. Can we remove this so we can provide specific exceptions back to the frontend ?

**Comment by powli** [comment]
I agree with @rikterbeek. In fact, in one of our payment extension we've created an around Plugin targeting `Magento\Checkout\Model\PaymentInformationManagement::savePaymentInformationAndPlaceOrder` for exactly that reason, so we could push the PaymentException message to the customer.

**Comment by joni-jones** [comment]
@rikterbeek, actually starting from Magento 2.2.4 release, you can map error messages from the Payment Gateway (see https://devdocs.magento.com/guides/v2.2/payments-integrations/payment-gateway/error-code-mapper.html).

> Can we remove this so we can provide specific exceptions back to the frontend?

It can't be removed for security reasons because not only payment integrations can throw errors.

**Comment by mcspronko** [comment]
@joni-jones @rikterbeek can this ticket be closed since we have support for error messages?
